LemonadeHomeTransition/ScrollableViewController.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  ScrollableViewController.swift
+//  LemonadeHomeTransition
+//
+//  Created by Dang Quoc Huy on 4/21/17.
+//  Copyright © 2017 Dang Quoc Huy. All rights reserved.
+//
+
+import UIKit
+
+protocol UIAnimateViewController {
+  var isFrom: Bool! { get set }
+  var index: Int! { get set }
+  
+  func setupInterpolationsFrom()
+  func animateFrom(progress: CGFloat)
+  func invalidateFrom()
+  
+  func setupInterpolationsTo()
+  func animateTo(progress: CGFloat)
+  func invalidateTo()
+}
+
+extension UIAnimateViewController {
+  
+  final func setupInterpolations(isFrom: Bool) {
+    isFrom ? setupInterpolationsFrom() : setupInterpolationsTo()
+  }
+  
+  final mutating func animate(progress: CGFloat, isFrom: Bool) {
```

Comment:
i don't even know what `final mutating` does 😅 . I thought these were for structs.